syncapi/synctypes/filter.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2017 Jan Christian Grünhage
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package synctypes
+
+import (
+	"errors"
+)
+
+// Filter is used by clients to specify how the server should filter responses to e.g. sync requests
+// Specified by: https://matrix.org/docs/spec/client_server/r0.5.0.html#filtering
```

Comment:
Should we update this reference to the new spec location?
https://spec.matrix.org/v1.6/client-server-api/#filtering